Paws4Me/Paws4Me/Data/LocalPetNameViewController.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  LocalPetNameViewController.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/10.
+//
+
+import CoreData
+import UIKit
+
+class LocalPetViewController: UIViewController, UITableViewDelegate {
+    // MARK: - IBOutlets
+    @IBOutlet weak private var petNameTableView: UITableView!
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+
+    // MARK: - Vars/Lets
+    private var namePet = ""
+    private var imagePet = ""
+    private var pets: [Pet]? = []
+
+    // MARK: - Life cycle
+    override func viewDidLoad() {
+      super.viewDidLoad()
+      petNameTableView.dataSource = self
+      petNameTableView.delegate = self
+      title = "Saved Pets"
+      fetchSavedPets()
+      savePets(name: namePet, image: imagePet)
+    }
+
+    func fetchSavedPets() {
+        do {
+            self.pets = try context?.fetch(Pet.fetchRequest())
+            DispatchQueue.main.async {
+                self.petNameTableView.reloadData()
+            }
+
+        } catch _ as NSError {
+            displayAlert(alertTitle: "Unable to retreive all your saved pets",
+                         alertMessage: "There was a problem retrieving",
+                         alertActionTitle: "Try again" ,
+                         alertDelegate: self, alertTriggered: .fatalLocalDatabaseAlert)
+        }
+    }
+
+    func savePets(name: String, image: String) {
+        guard let petContext = self.context else { return }
+        let petObject = Pet(context: petContext)
+        petObject.petName = namePet
+        petObject.petImage = imagePet
+
+      do {
+          guard let petContext = self.context else { return }
```

Comment:
Code like this doesn't belong in the `ViewController`

Reply:
I definitely agree! This was fixed when I merged develop into this branch